pkg/koordlet/metricsadvisor/collector.go

```diff
@@ -253,6 +281,15 @@ func (c *collector) collectPodResUsed() {
 				MemoryWithoutCache: *resource.NewQuantity(memUsageValue, resource.BinarySI),
 			},
 		}
+		if c.haveGPU {
```

Comment:
seems too many checks of `if c.haveGPU` in collector main process.
consider define `GetPodGPUUsage(podMetric.GPUs, ...)`, `GetContainerGPUUsage(container.GPUs...)`, `GetNodeGPUUsage(node.GPUs...)` inside gpuDeviceManager, gpuDeviceManager can ignore the collection if the node does not have gpu.
so that gpu collection can be integrated together inside.